### PR TITLE
Add Data docs index page

### PR DESCRIPTION
**Pull request by megcrenshaw**
Builds on https://github.com/code-dot-org/code-dot-org/pull/48281 by finishing the student/teacher-facing Data Docs index page:

Screen Shot 2022-09-26 at 2 02 35 PM

## Testing story

I'm not sure I love these unit tests –– they rely heavily on the implementation

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/dataDocs/DataDocIndex.jsx
Comment (megcrenshaw) [inline apps/src/templates/dataDocs/DataDocIndex.jsx:7]: Checked with Tess: translation is not happening yet

**Comment by dmcavoy** [comment]
Is there a reason we changed the design for this page? It looks like in the spec: https://docs.google.com/document/d/1fIiODFYU1J0NCfrK-Jt27Iy3zyMoPtn3WZAq35IrU80/edit#heading=h.8n353qbx5ec7 we had planned for this page to use the same styling that we use for Reference Guides and Code Docs.

**Comment by megcrenshaw** [comment]
> Is there a reason we changed the design for this page? It looks like in the spec: https://docs.google.com/document/d/1fIiODFYU1J0NCfrK-Jt27Iy3zyMoPtn3WZAq35IrU80/edit#heading=h.8n353qbx5ec7 we had planned for this page to use the same styling that we use for Reference Guides and Code Docs.

Good question –– we decided *not* to have a navigation bar (it's been struck out in the [engineering plan](https://docs.google.com/document/d/1GDGpuZWjhS9y96rCFqCk8uLJD9SzoZUiLCIry6WdHEI/edit#heading=h.p3guq9bi2b0s)), and instead just list them out. Neither Reference Guides nor Programming Expressions has this pattern –– do you have ideas on what this page should look like instead? The instructions I got were to keep this as simple as possible, but if you have a sense of what might make it look nicer, let me know

**Comment by dmcavoy** [comment]
I was thinking it would look like these pages in reference guides and code documentation. Tim made the navigation bar component reusable. 

https://levelbuilder-studio.code.org/programming_environments/spritelab
https://levelbuilder-studio.code.org/courses/csp-2022/guides/game-lab-overview

If the plan changed between you all and Tess thats totally fine with me.

**Comment by megcrenshaw** [comment]
Yeah, the nav bar in Ref Guides (and Prog Exp) uses categories, which we don't have, and which has different logic for the order in which they're shown.

I'm down to use something that's been used, but a basic list for the index page I didn't see an example for

**Comment by davidsbailey** [comment]
I do vaguely recall saying we were going to skip the left nav, because it was going to save a big chunk of work. however it'd be great to get buy-in from a PM on what the index page should look like in this case. @tess323 do the image above look ok to you?

**Comment by tess323** [comment]
I am okay with this as is.  Cutting the nav bar still feels like the right move to me and since these are linked from the documentation itself I do not expect to see much traffic on this page.

**Comment by davidsbailey** [comment]
thank you @tess323 ! Can you remind me where (if anywhere) we might actually be linking to this "all docs" page from? naively, it seems like the "data library" tab in applab is what people will be using to see all docs.

File: apps/test/unit/templates/dataDocs/DataDocIndexTest.js
Comment (davidsbailey) [inline apps/test/unit/templates/dataDocs/DataDocIndexTest.js:49]: tests look great to me! simple component --> simple test(s)

**Review by davidsbailey — APPROVED** [review]
LGTM once backend PR is merged

**Comment by megcrenshaw** [comment]
> thank you @tess323 ! Can you remind me where (if anywhere) we might actually be linking to this "all docs" page from? naively, it seems like the "data library" tab in applab is what people will be using to see all docs.

Yep, just chatted with Tess about this –– we link to individual data docs through applab (see photo below), but we're not linking to the index page from anywhere

image

File: apps/test/unit/templates/dataDocs/DataDocIndexTest.js
Comment (megcrenshaw) [reply apps/test/unit/templates/dataDocs/DataDocIndexTest.js:49]: I added a couple tests to make it clear we don't show the docs if they don't have both a name and content

**Pull request merged**
